cmd/gb-vendor/vendor/repo.go

```diff
@@ -137,13 +134,22 @@ func (g *gitrepo) URL() string {
 	return g.url
 }
 
-func (g *gitrepo) Clone() (WorkingCopy, error) {
+func (g *gitrepo) Checkout(branch, revision string) (WorkingCopy, error) {
 	dir, err := mktmp()
 	if err != nil {
 		return nil, err
 	}
 
-	if err := runOut(os.Stderr, "git", "clone", g.url, dir); err != nil {
+	args := []string{
+		"clone",
+		g.url,
+		dir,
+	}
+	if branch != "" {
+		args = append(args, "--branch", branch)
```

Comment:
>
> Nevertheless, you can checkout either branch or commit. You can't do both
> at once. User can't specify branch and revision at once, there must be a
> check if user didn't do that.
>
I think you can do both on mercurial, commits are related to branches, ie
some hash may only be valid on a specific branch.


>  If you want to checkout commit (revision) you can do something like this:
>
> git clone $repo
> git reset --hard $revision
>
> You know a lot more about this than I do, do you want to do a change when
#191 lands ?




> —
> Reply to this email directly or view it on GitHub
> <https://github.com/constabulary/gb/pull/191/files#r31875111>.
>